app/models/vote.rb

```diff
@@ -0,0 +1,8 @@
+class Vote < ApplicationRecord
+  validates :user_id, presence: true
+  validates :work_id, presence: true
```

Comment:
You don't need to have these, the relationships require them anyway.

However you **do** need a validation to prevent a duplicate combination user_id and work_id.

```suggestion
   #  Requires every combination work_id and user_id to be unique.
  validates :user_id, uniqueness: {scope: :work_id}
```